### ISSUE TITLE
Unable to load MNIST dataset in the "func prepareBatchProvider()" in MNIST.swift

### ISSUE THREAD

**Issue by kimihe** [JacopoMangiavacchi/MNIST-CoreML-Training#2]
Dear Mangiavacchi,

I am a Ph.D. student working in on-device learning, surprisingly finding your **MNIST-CoreML-Training** repo is a valuable guide for our academic research. I downloaded the codes and tried to run the app on my iPhone.
However, I found there may be some problems in the `func prepareBatchProvider() -> MLBatchProvider` in `MNIST.swift`.
I will present the relevant snippet first.

```
let trainFilePath = Bundle.main.url(forResource: "mnist_train", withExtension: "csv")!
            if freopen(trainFilePath.path, "r", stdin) == nil {
                print("error opening file")
            }
            while let line = readLine()?.split(separator: ",") {
                count += 1
                DispatchQueue.main.async {
                    self.trainingBatchStatus = .preparing(count: count)
                }

                let imageMultiArr = try! MLMultiArray(shape: [1, 28, 28], dataType: .float32)
                let outputMultiArr = try! MLMultiArray(shape: [1], dataType: .int32)

                for r in 0..<28 {
                    for c in 0..<28 {
                        let i = (r*28)+c
                        imageMultiArr[i] = NSNumber(value: Float(String(line[i + 1]))! / Float(255.0))
                    }
                }
```
Following the description in your blog, this snippet is for MNIST batch images loading and data normalization. However, the line ` imageMultiArr[i] = NSNumber(value: Float(String(line[i + 1]))! / Float(255.0))` depends on the variable `line[]` and `line[]` is just the array of the `mnist_train.csv` file (the same to `mnist_test.csv`), not the images of MNIST. This is the major problem I faced.

As the MNIST dataset is used during the training on the phone, I guess we need to download the dataset or directly package it in the app (like what we usually do when handling the training on PC). **Could you help me address this problem or tell me how to correctly load the MNSIT on iPhone?**

 Thank you very much!

**Comment by florentmorin** [comment]
The problem is due to LFS.

When trying to fetch data via `git lfs fetch origin master`, you'll obtain this message:
```
fetch: Fetching reference refs/heads/master
batch response: This repository is over its data quota. Account responsible for LFS bandwidth should purchase more data packs to restore access.
error: failed to fetch some objects from 'https://github.com/JacopoMangiavacchi/MNIST-CoreML-Training.git/info/lfs'
```

To avoid need of LFS, I just created a version including CSV files compressed with LZFSE. #3

**Comment by JacopoMangiavacchi** [comment]
@florentmorin Thanks for the PR and sorry again for the delay approving it.
